sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -244,3 +264,61 @@ def _test_pipeline_workflow(self,
         self.assertEqual(golden, compiled)
     finally:
       shutil.rmtree(temp_dir)
+
+  def _test_workflow_without_decorator(self, pipeline_yaml, params_dict):
+    """Test compiling a workflow and appending pipeline params."""
+
+    test_data_dir = os.path.join(os.path.dirname(__file__), 'testdata')
+    golden_yaml_file = os.path.join(test_data_dir, pipeline_yaml)
+    temp_dir = tempfile.mkdtemp()
+    try:
+      compiled_workflow = compiler.TektonCompiler()._create_workflow(
+          params_dict['function'],
+          params_dict.get('name', None),
+          params_dict.get('description', None),
+          params_dict.get('paramsList', None),
+          params_dict.get('conf', None))
+      if GENERATE_GOLDEN_YAML:
```

Comment:
There is a lot of duplicated code in the last 3 methods, but I don't see a nice way to refactor it.
Maybe we can at least factor out this part:

```Python
    test_data_dir = os.path.join(os.path.dirname(__file__), 'testdata')
    golden_yaml_file = os.path.join(test_data_dir, pipeline_yaml)
...
      if GENERATE_GOLDEN_YAML:
        with open(golden_yaml_file, 'w') as f:
          yaml.dump_all(compiled_workflow, f, default_flow_style=False)
      else:
        with open(golden_yaml_file, 'r') as f:
          golden = list(yaml.safe_load_all(f))
        self.maxDiff = None
        self.assertEqual(golden, compiled_workflow)
```

into a funtion called `def verify_compiled_workflow(pipeline_yaml, compiled_workflow)`